src/Identification/Queue/Jobs/Job.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the tenancy/tenancy package.
+ *
+ * Copyright Tenancy for Laravel
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @see https://tenancy.dev
+ * @see https://github.com/tenancy
+ */
+
+namespace Tenancy\Identification\Drivers\Queue\Jobs;
+
+use ReflectionClass;
+
+class Job
+{
+    protected $tenant;
+
+    protected $tenant_identifier;
+
+    protected $tenant_key;
+
+    public function getTenant()
+    {
+        return $this->tenant;
+    }
+
+    public function getTenantIdentifier()
+    {
+        return $this->tenant_identifier;
+    }
+
+    public function getTenantKey()
+    {
+        return $this->tenant_key;
+    }
+
+    public function __unserialize(array $values)
+    {
+        $properties = (new ReflectionClass($this))->getProperties();
+
+        $class = get_class($this);
+
+        foreach ($properties as $property) {
+            if (!in_array($property->getName(), ['tenant', 'tenant_identifier', 'tenant_key'])) {
+                continue;
+            }
+
+            $name = "\0*\0{$property->getName()}";
+
+            if (!array_key_exists($name, $values)) {
+                continue;
+            }
+
+            $property->setAccessible(true);
+
+            $property->setValue($this, unserialize(serialize($values[$name])));
```

Comment:
As the value might now be an incomplete PHP Class, we serialize it and then unserialize it again to get the actual value :)

Reply:
I'm sure it works, and understand what it's doing from your description, but it reads really weird